rocAL/rocAL/source/loaders/image/image_read_and_decode.cpp

```diff
@@ -132,6 +134,29 @@ ImageReadAndDecode::set_batch_random_bbox_crop_coords(std::vector<std::vector<fl
     _crop_coords_batch = crop_coords;
 }
 
+void ImageReadAndDecode::feed_external_input(std::vector<std::string> input_images_names, std::vector<int> labels, std::vector<unsigned char *> input_buffer,
+                             std::vector<unsigned> roi_width, std::vector<unsigned> roi_height,
+                             unsigned int max_width, unsigned int max_height, int channels,  FileMode mode, bool eos)
+{
+    std::vector<size_t> image_size; // check if its required
+    image_size.reserve(roi_height.size());
+    for(unsigned int i = 0; i < roi_height.size(); i++) {
+        if (mode == 2)
+            image_size[i] = (roi_width[i] * roi_height[i] * channels);
+        else if (mode == 1)
+            image_size[i] = roi_height[i];
+    }
+    if(mode == 0) {
+        _reader->feed_file_names(input_images_names, input_images_names.size(), eos);
+    }
+    else if(mode == 1) {
+        _reader->feed_data(input_buffer, image_size, mode, eos, max_width, max_height, 3);
+    }
+    else if(mode == 2){
+        _reader->feed_data(input_buffer, image_size, mode, eos, max_width, max_height, 3);
+    }
```

Comment:
what is the difference between mode 1 and mode 2
both are exactly calling the same feed_data call with same params

If the calls are same you can merge both of them in single branch statement